infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -48,6 +47,7 @@
     private ExecutorService processorService = Executors
             .newSingleThreadExecutor(new ThreadFactoryBuilder()
                     .setDaemon(false)
+                    .setPriority(Thread.MAX_PRIORITY)
```

Comment:
Unfortunately, in reality, it doesn't give us any advantages, setting this parameter doesn't affect anything https://www.javamex.com/tutorials/threads/priority_what.shtml

Anyway, good to have it

Reply:
According the article, it does work in linux, just need to run as root and enable the jvm parameter.

Reply:
>Under Linux, you have to go through more hoops to get thread priorities to function at all

>for them to work, you must be running as root (or with root privileges via setuid);
which is not our case

>the JVM parameter -XX:UseThreadPriorities must be included.

>And does anyone really run, say, their web server as root?

>The graph shows that there is a correlation between Java priority (nice value) and CPU allocation, although it is far from linear.

So, in our case, no effect at all

Reply:
> the JVM parameter -XX:UseThreadPriorities must be included.
> And does anyone really run, say, their web server as root?
It depends on how you deploy, for example corfu is deployed with nice value -10 in nsx. Also giving root privilege to a service is not uncommon. Anyway it doesn't contradict with this patch, it depends on how you deploy.

> The graph shows that there is a correlation between Java priority (nice value) and CPU allocation, although it is far from linear.
Of course it cannot be linear, also nice value is just a hint, but it does work to some extent.